agent/src/configuration_wrapper/configuration_wrapper.rs

```diff
@@ -0,0 +1,31 @@
+use config::{Config, ConfigError, Environment, File};
+
+#[derive(Debug)]
+pub struct ConfigurationWrapper {
+    config: Config,
+}
+
+impl ConfigurationWrapper {
+    pub fn new() -> Result<Self, ConfigError> {
```

Comment:
care new is a key word in rust

Reply:
no it's not, see: [https://doc.rust-lang.org/reference/keywords.html](https://doc.rust-lang.org/reference/keywords.html)